### PR TITLE
backupccl: Cosmetic changes to backup_processor.go to improve clarity.

### PR DESCRIPTION
**Pull request by benbardin**
Release note: None

Renames a few things, and moves the sstsink into its own file.

### CODE REVIEW

**Comment by benbardin** [comment]
Required tests are passing!

File: pkg/ccl/backupccl/backup_processor.go [hunk 1]
```diff
@@ -355,7 +347,7 @@ func runBackupProcessor(
 						splitMidKey = true
 					}
 
-					req := &roachpb.ExportRequest{
+					KVReq := &roachpb.ExportRequest{
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:350]: we usually lead with lowercase for local vars, since otherwise they look like exported package vars, so e.g. `kvReq`.  

the name seems a tad weird to me, since the only "requests" we ever make are kv requests, and they're typically just called "req" throughout the codebase (e.g. `git  grep "req :=.*roachpb.*Request"`), but it is technically correct I suppose?

File: pkg/ccl/backupccl/backup_processor.go [hunk 2]
```diff
@@ -514,34 +506,34 @@ func runBackupProcessor(
...
+					if len(KVResponse.Files) > 1 {
 						log.Warning(ctx, "unexpected multi-file response using header.TargetBytes = 1")
 					}
 
-					for i, file := range res.Files {
-						f := BackupManifest_File{
+					for i, file := range KVResponse.Files {
+						partialFile := BackupManifest_File{
 							Span:        file.Span,
-							Path:        file.Path,
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:524]: IMO I'd make a separate commit here that deletes the `Path` field from ExportResponse_File (and reserves it) just to prove it was never being set, which makes it obvious that deleting this line is safe/correct (which I think it is)

File: pkg/ccl/backupccl/backup_processor.go [hunk 3]
```diff
@@ -247,9 +239,9 @@ type spanAndTime struct {
 	lastTried  time.Time
 }
 
-type returnedSST struct {
-	f              BackupManifest_File
-	sst            []byte
+type returnedSpans struct {
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:242]: I think this should be singular?

File: pkg/ccl/backupccl/backup_processor.go [hunk 4]
```diff
@@ -247,9 +239,9 @@ type spanAndTime struct {
...
+	partialFile    BackupManifest_File
+	spanDataAsSST  []byte
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:244]: Go generally discourages repeated word or "stuttering" names, e.g. preferring "user.Email" over "user.UserEmail", and letting the containing type/package/etc speak for itself, so in this case I think it'd be "returnedSpan.dataSST" since it is the data field _of_ the returnedSpan.

File: pkg/ccl/backupccl/backup_processor.go [hunk 5]
```diff
@@ -471,23 +463,23 @@ func runBackupProcessor(
 						return errors.Wrapf(exportRequestErr, "exporting %s", span.span)
 					}
 
-					res := rawRes.(*roachpb.ExportResponse)
+					KVResponse := rawKVResponse.(*roachpb.ExportResponse)
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:466]: ditto above, we _usually_ just call this `resp`, but if we are going to `kv` prefix it here, we'd do so lower case.

File: pkg/ccl/backupccl/backup_processor.go [hunk 6]
```diff
@@ -514,34 +506,34 @@ func runBackupProcessor(
...
 
-					if len(res.Files) > 1 {
+					if len(KVResponse.Files) > 1 {
 						log.Warning(ctx, "unexpected multi-file response using header.TargetBytes = 1")
 					}
 
-					for i, file := range res.Files {
-						f := BackupManifest_File{
+					for i, file := range KVResponse.Files {
+						partialFile := BackupManifest_File{
```
Comment (dt) [inline pkg/ccl/backupccl/backup_processor.go:514]: calling this a "file" at all is a tad weird; it is just a some data and metadata that identified the span of keys and time that data came from and how much is in it, and isn't a "file" at all just yet, partial or otherwise (it could be complete when we go to write it, for all we know). it could be less confusing to avoid the extra local var completely here: make the `ret` returnedSpan here, _embed_ Manifest_File in it with a comment saying "BackupManifest_File conveniently already has all the metadata fields we need to identify a key/time span" and avoid the word file in this block entirely?

File: pkg/ccl/backupccl/backup_processor.go (hunk 3 above)
Comment (dt) [reply pkg/ccl/backupccl/backup_processor.go:242]: maybe should be called "exportedSpan" actually; "returnedSpan" just sounds like somebody returned a roachpb.Span, but this is much more than that; it is the actual data from that span.

File: pkg/ccl/backupccl/file_sst_sink.go [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+		spanGrows   int
+	}
+
+	memAcc struct {
+		ba            *mon.BoundAccount
+		reservedBytes int64
+	}
+}
+
+func makeFileSSTSink(
```
Comment (dt) [inline pkg/ccl/backupccl/file_sst_sink.go:76]: I'm not sure that "fileSSTSink" is really telling me anything that "sstSink" didn't, but I also sympathize that it wasn't telling much / enough before.

I think the distinguishing characteristics of this thing are that it takes chunks of exported data (which just so happens to be in sst format), buffers and merges it, and then writes SSTables. So maybe bufferedSSTWriter or something like that?

File: pkg/ccl/backupccl/backup_processor.go (hunk 2 above)
Comment (benbardin) [reply pkg/ccl/backupccl/backup_processor.go:524]: Sure, reverted for now.

File: pkg/ccl/backupccl/backup_processor.go (hunk 3 above)
Comment (benbardin) [reply pkg/ccl/backupccl/backup_processor.go]: Changed, but you sure about singular? There's literally a field called `completedSpans` in the object.

File: pkg/ccl/backupccl/backup_processor.go (hunk 6 above)
Comment (benbardin) [reply pkg/ccl/backupccl/backup_processor.go]: Yeah I like that! How's this look?

File: pkg/ccl/backupccl/file_sst_sink.go (hunk 7 above)
Comment (benbardin) [reply pkg/ccl/backupccl/file_sst_sink.go:76]: To me, calling the object a `FileSSTSink` highlights that it's not a generic sst sink. It outputs `BackupManifest_File` objects to the progress channel as one of two core outputs, and takes in a `BackupManifest_File` as an input in `exportedSpan` (though we're now more subtly calling that `metadata`). This isn't something that can be used to buffer any other kind of object, which took me a few hours to totally figure out.

Have I convinced?

**Review by dt — APPROVED** [review]
